Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -9246,6 +9246,111 @@ def get_template_stacks(
+    """
+    Gets global counter information from all the PAN-OS firewalls in the topology
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: ShowCounterGlobalCommmandResult = FirewallCommand.get_counter_global(topology, device_filter_string)
+    return result
+
+
+def get_bgp_peers(topology: Topology, device_filter_string: str = None) -> ShowRoutingProtocolBGPCommandResult:
+    """
+    Retrieves all BGP peer information from the PAN-OS firewalls in the topology.
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: ShowRoutingProtocolBGPCommandResult = FirewallCommand.get_bgp_peers(topology, device_filter_string)
+    return result
+
+
+def get_device_connectivity(topology: Topology, hostid: str) -> GetDeviceConnectivityCommandResult:
+    """
+    Search the topology for a given hostid. If it's not found, the device is not connected or unreachable.
+    :param topology: `Topology` instance !no-auto-argument
+    :param hostid: ID (serial or hostname) of host to search for.
+    :return:
+    """
+    if not topology.get_by_filter_str(hostid):
+        result = GetDeviceConnectivityResultData(
+            hostid=hostid,
+            connected=False
+        )
+        return GetDeviceConnectivityCommandResult(summary_data=[result])
+    else:
+        result = GetDeviceConnectivityResultData(
+            hostid=hostid,
+            connected=True
+        )
+        return GetDeviceConnectivityCommandResult(summary_data=[result])
+
+
+def get_available_software(topology: Topology, device_filter_string: str = None) -> SoftwareVersionCommandResult:
+    """
+    Check the devices for software that is available to be installed.
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: SoftwareVersionCommandResult = UniversalCommand.get_available_software(topology, device_filter_string)
+    return result
+
+
+def get_ha_state(topology: Topology, device_filter_string: str = None) -> List[ShowHAState]:
+    """
+    Get the HA state and associated details from the given device and any other details.
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: List[ShowHAState] = FirewallCommand.get_ha_status(topology, device_filter_string)
+    return result
+
+
+def get_jobs(topology: Topology, device_filter_string: str = None, status: str = None, job_type: str = None,
+             id: int = None) -> List[ShowJobsAllResultData]:
```

Comment:
```suggestion
             id: Optional[str] = None) -> List[ShowJobsAllResultData]:
```

By default the arguments that are being passed from ```demisto.args()``` are defined as strings, hence its a string and not an integer.

There are build-in functions in common server python which allows you to change that string if required to another type. 

For example - '1' as a string to 1 as an integer, or 'true' as a string to a True as a boolean, or a list of strings separated by a comma, 'value1,value2' to a list of ['value1', value2'].

Reply:
Yep, true - fixed in all instances that this occured i think